ocaml-lsp-server/src/code_actions/action_type_annotate.ml

```diff
@@ -59,6 +103,35 @@ let code_action_of_type_enclosing uri doc (loc, typ) =
     ~isPreferred:false
     ()
 
+let code_action_of_type_enclosing' uri doc (loc, arg_len, typ) =
+  let open Option.O in
+  let+ original_text = get_source_text doc loc in
+  let arrow = " -> " in
+  let typ' =
+    Str.split (Str.regexp arrow) typ
```

Comment:
`Str` isn't thread safe so we avoid it. Try using `Re.split` or something similar.

Reply:
It's also not ideal that we need to deal with strings here at all btw. Shouldn't it be possible to use the type tree representation to drop the args instead?

Reply:
Hey Rudi, it's just a draft and work in progress. I have opened a PR so that it's visible that I'm working on this feature.
Thanks for your suggestions (will keep them in mind), but I'm not sure if it's worth it to do review now. 🙂

Reply:
Sure thing. Some people like the feedback early, others prefer it all at the end. I can go either way